unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,235 @@
+            redis_client_auto_register.is_active = False
+            redis_client_auto_register.unregister(is_blocking=True)
+
+    @staticmethod
+    @patch("d3a_api_client.redis_client_base.wait_until_timeout_blocking",
+           side_effect=AssertionError)
+    def test_unregister_is_active_true_throws_exception(
+            mock_wait_until_timeout_blocking,
+            redis_client_auto_register):
+        """Check if is active set to true throws exception."""
+        with pytest.raises(RedisAPIException,
+                           match="API unregister process timed out. "
+                                 "Server will continue processing "
+                                 "your request on the background and "
+                                 "will notify you as soon as "
+                                 "the unregistration has been completed."):
+            redis_client_auto_register.is_active = True
+            redis_client_auto_register.unregister(is_blocking=True)
+        mock_wait_until_timeout_blocking.assert_called()
+
+    @staticmethod
+    @patch("uuid.uuid4", return_value="some-transaction-uuid")
+    def test_on_register(mock_uuid, redis_client_auto_register):
+        """Check the on_register function with correct message that doesn't throw exception."""
+        redis_client_auto_register.register(is_blocking=False)
+        data = {"name": AREA_ID, "device_uuid": DEVICE_ID, "transaction_id": mock_uuid()}
+        message = {"data": json.dumps(data)}
+        redis_client_auto_register._on_register(message)
+        assert redis_client_auto_register.is_active is True
+        assert redis_client_auto_register.area_uuid == DEVICE_ID
+
+    @staticmethod
+    @patch("uuid.uuid4", return_value="some-transaction-uuid")
+    def test_on_unregister(mock_uuid, redis_client_auto_register):
+        """Check the on_unregister function with correct message that doesn't throws exception."""
+        redis_client_auto_register.is_active = True
+        redis_client_auto_register.unregister(is_blocking=False)
+        data = {"name": AREA_ID, "device_id": DEVICE_ID,
+                "transaction_id": mock_uuid(), "response": "success"}
+        message = {"data": json.dumps(data)}
+        redis_client_auto_register._on_unregister(message)
+        assert redis_client_auto_register.is_active is False
+
+    @staticmethod
+    @patch("uuid.uuid4", return_value="some-transaction-uuid")
+    def test_on_unregister_throws_exception(mock_uuid, redis_client_auto_register):
+        """Check if exception is raised when response of on_unregister is not successful."""
+        with pytest.raises(RedisAPIException,
+                           match=f"Failed to unregister from market {AREA_ID}."
+                                 "Deactivating connection."):
+            redis_client_auto_register.is_active = True
+            redis_client_auto_register.unregister(is_blocking=False)
```

Comment:
Same as above

Reply:
I've answered above in this comment: https://github.com/gridsingularity/d3a-api-client/pull/163#discussion_r721442048